### ISSUE TITLE
Unable to write error log if ConductR CLI home directory is not yet created

### ISSUE THREAD

**Issue by fsat** [typesafehub/conductr-cli#180]
CLI will attempt to write to error log file stored in the ConductR CLI home directory when encountering error.

Depending what command is run and when the error occurs, this directory won't always be created. In the example below, the `conduct version` command is run before anything else is run. An error is raised which can't be logged since the home directory is not yet created.

```
ubuntu@ip-10-0-1-200:~$ conduct version
Encountered unexpected error.
Reason: AttributeError 'Namespace' object has no attribute 'dcos_mode'
Further information of the error can be found in the error log file: /home/ubuntu/.conductr/errors.log
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/conductr_cli/main_handler.py", line 8, in run
    result = callback()
  File "/usr/local/lib/python3.5/dist-packages/conductr_cli/conduct.py", line 6, in main_method
    conduct_main.run()
  File "/usr/local/lib/python3.5/dist-packages/conductr_cli/conduct_main.py", line 368, in run
    args.cli_parameters = get_cli_parameters(args)
  File "/usr/local/lib/python3.5/dist-packages/conductr_cli/conduct_main.py", line 286, in get_cli_parameters
    if not args.dcos_mode:
AttributeError: 'Namespace' object has no attribute 'dcos_mode'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/conduct", line 11, in <module>
    sys.exit(run())
  File "/usr/local/lib/python3.5/dist-packages/conductr_cli/conduct.py", line 10, in run
    main_handler.run(main_method)
  File "/usr/local/lib/python3.5/dist-packages/conductr_cli/main_handler.py", line 31, in run
    handler = logging.handlers.RotatingFileHandler(DEFAULT_ERROR_LOG_FILE, maxBytes=3000000, backupCount=1)
  File "/usr/lib/python3.5/logging/handlers.py", line 150, in __init__
    BaseRotatingHandler.__init__(self, filename, mode, encoding, delay)
  File "/usr/lib/python3.5/logging/handlers.py", line 57, in __init__
    logging.FileHandler.__init__(self, filename, mode, encoding, delay)
  File "/usr/lib/python3.5/logging/__init__.py", line 1008, in __init__
    StreamHandler.__init__(self, self._open())
  File "/usr/lib/python3.5/logging/__init__.py", line 1037, in _open
    return open(self.baseFilename, self.mode, encoding=self.encoding)
FileNotFoundError: [Errno 2] No such file or directory: '/home/ubuntu/.conductr/errors.log'
```

**Comment by fsat** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [ConductR CLI: failure to log if CLI home directory is not created](https://trello.com/c/7lQBjMA4/380-conductr-cli-failure-to-log-if-cli-home-directory-is-not-created)

**Comment by fsat** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [ConductR CLI: Fixes prior 2.0.0-RC1](https://trello.com/c/AXKwNWwh/379-conductr-cli-fixes-prior-2-0-0-rc1)
